### PR TITLE
#67: Add Data Annotations to Request Models

### PR DESCRIPTION
**Pull request by amrwc**

### CODE REVIEW

File: src/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (amrwc) [inline src/.idea/.gitignore]: All the files in `.idea` have been auto generated by the Rider IDE.

**Comment by amrwc** [comment]
Pro tip: tick the `Hide whitespace changes` because some of the lines have just been reshuffled.
image

File: src/Smp.Web/Controllers/AccountsController.cs [hunk 2]
```diff
@@ -31,24 +33,25 @@ public AccountsController(IUsersRepository usersRepository, IActionsRepository a
         public async Task<IActionResult> ForgottenPassword(string email)
         {
             var user = await _usersRepository.GetUserByEmail(email);
-            if (user == null) return NotFound(new Error("invalid_email", "Email must be linked to an existing user account."));
-
+            if (user == null) return NotFound(new Error("invalid_email",
```
Comment (faibz) [inline src/Smp.Web/Controllers/AccountsController.cs]: Probably best to just put the return on a new line.

File: src/Smp.Web/Controllers/PostsController.cs [hunk 3]
```diff
@@ -1,43 +1,41 @@
-using System;
```
Comment (faibz) [inline src/Smp.Web/Controllers/PostsController.cs]: What happened here?

File: src/Smp.Web/Services/AccountsService.cs [hunk 4]
```diff
@@ -1,6 +1,5 @@
 ﻿using System;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
```
Comment (amrwc) [inline src/Smp.Web/Services/AccountsService.cs]: Unused import.

File: src/Smp.Web/Controllers/PostsController.cs [hunk 5]
```diff
@@ -1,43 +1,41 @@
...
+        public async Task<IActionResult> CreatePost([FromBody] CreatePostRequest createPostRequest)
+        {
+            var tkn = Request.Headers["Authorization"];
+            var authorized = _authService.AuthorizeSelf(tkn, createPostRequest.SenderId) &&
+                (_authService.AuthorizeSelf(tkn, createPostRequest.ReceiverId) ||
+                    await _authService.AuthorizeFriend(tkn, createPostRequest.ReceiverId));
+            if (!authorized) return Unauthorized();
+            await _postsService.CreatePost(new Post(createPostRequest));
+            return Ok();
+        }
```
Comment (faibz) [inline src/Smp.Web/Controllers/PostsController.cs]: While I don't intend on trying to enforce anything, I think it's reasonable for us to use whitespace to split some things up logically.

File: src/Smp.Web/Smp.Web.csproj [hunk 6]
```diff
@@ -1,83 +1,83 @@
...
+    <!-- As part of publishing, ensure the JS resources are freshly built in production mode -->
+    <Exec WorkingDirectory="$(SpaRoot)" Command="npm install" />
+    <Exec WorkingDirectory="$(SpaRoot)" Command="npm run build -- --prod" />
+    <Exec WorkingDirectory="$(SpaRoot)" Command="npm run build:ssr -- --prod" Condition=" '$(BuildServerSideRenderer)' == 'true' " />
+
+    <!-- Include the newly-built files in the publish output -->
+    <ItemGroup>
+      <DistFiles Include="$(SpaRoot)dist\**; $(SpaRoot)dist-server\**" />
+      <DistFiles Include="$(SpaRoot)node_modules\**" Condition="'$(BuildServerSideRenderer)' == 'true'" />
+      <ResolvedFileToPublish Include="@(DistFiles->'%(FullPath)')" Exclude="@(ResolvedFileToPublish)">
```
Comment (amrwc) [inline src/Smp.Web/Smp.Web.csproj]: I guess only VS Code prefers `&gt;`. ¯\\_(ツ)_/¯

Comment (amrwc) [inline src/Smp.Web/Smp.Web.csproj]: I guess only VS Code prefers `&gt;`. ¯\\_(ツ)_/¯

File: src/Smp.Web/Models/Requests/RequestRequest.cs [hunk 7]
```diff
@@ -5,11 +5,8 @@ namespace Smp.Web.Models.Requests
 {
     public class RequestRequest
     {
-        [Required]
-        public Guid SenderId { get; set; }
-        [Required]
-        public Guid ReceiverId { get; set; }
-        [Required]
-        public byte RequestTypeId { get; set; }
+        [Required] public Guid SenderId { get; set; }
```
Comment (faibz) [inline src/Smp.Web/Models/Requests/RequestRequest.cs]: Please pick one style. Having attributes on a separate line above the property seems to be the most used way of doing this.

File: src/Smp.Web/Controllers/PostsController.cs (hunk 3 above)
Comment (amrwc) [reply src/Smp.Web/Controllers/PostsController.cs]: I think it's one of those `^M` issues, aka the difference in how macOS and Windows mark new line/line feed. I remember seeing it in a diff.

File: src/Smp.Web/Models/Requests/RequestRequest.cs (hunk 7 above)
Comment (amrwc) [reply src/Smp.Web/Models/Requests/RequestRequest.cs]: Notice how those on a separate lines are way longer, therefore the lines would exceed the 120 character limit enforced by the style checker. I'll revert to having the attributes on a separate line and amend the style checker to enforce this style in the future.

Comment (amrwc) [reply src/Smp.Web/Models/Requests/RequestRequest.cs]: Current values:
image

Comment (amrwc) [reply src/Smp.Web/Models/Requests/RequestRequest.cs]: Addressed in 9902032.

File: src/Smp.Web/Controllers/PostsController.cs (hunk 5 above)
Comment (amrwc) [reply src/Smp.Web/Controllers/PostsController.cs]: Addressed in 9902032.

File: src/Smp.Web/Controllers/AccountsController.cs (hunk 2 above)
Comment (amrwc) [reply src/Smp.Web/Controllers/AccountsController.cs]: Addressed in 9902032.

**Pull request merged**